<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>H. Papilio DAPC, 32 Sequence Reference</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Population Genomics in Arcellinida</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="papilio_32.html">H. Papilio, 32 Sequence Reference, Dendrogram &amp; PCA</a>
</li>
<li>
  <a href="papilio_32_dapc.html">H. Papilio, 32 Sequence Reference, DAPC</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">H. Papilio DAPC, 32 Sequence Reference</h1>
<h4 class="date">1/7/2021</h4>

</div>


<p>We will be working with a genlight object for this analysis, so we convert the vcfR object to a genlight and add the population information.</p>
<p>Discriminant analysis of principal components is a multivariate approach that divides variance into within-group variance and between-group variance to discriminate between groups. This enables the identification of genetic clusters, and the clusters identified by the model can be compared by those expected from geographic information. This provides a quality alternative for clustering without making assumptions about the randomness of mating, which may be unknown in microbial species, as is the case with <em>H. papilio</em>.</p>
<p>To perform DAPC analysis, the first step is to determine the optimal number of clusters. Once we do so, we can compare the detected clusters with the geography of the samples.</p>
<pre class="r"><code>### DAPC analysis

#find optimum number of clusters
#run this line of code without the n.pca and n.clust arguments, and use the plot to identify the best values for these arguments
#for number of clusters, look for local minimum 
#for number of PCAs, look at point at which additional PCAs do not explain a significant amount of variance
clust &lt;- find.clusters(pap_76_genlight, max.n.clust = 15, n.pca = 40, n.clust = 7)

#compare to the actual populations
table(pop(pap_76_genlight), clust$grp )</code></pre>
<pre><code>##                         
##                          1 2 3 4 5 6 7
##   Acadia National Park   3 1 3 8 2 0 0
##   Great Cranberry Island 0 0 0 1 0 0 0
##   Harvard Forest         0 0 5 0 0 0 0
##   Hawley bog             0 4 1 0 2 2 7
##   Orono bog              0 0 4 0 3 1 0</code></pre>
<p>With the cluster information, we can perform the DAPC analysis with the dapc() function.</p>
<p>This allows us to create a memborship probability plot, where we compare the populations assigned by the clustering analysis done by DAPC with the geographical information about the samples. Here, we can see a high degree of heterogeneity in each location, indicating low genetic differentiation between populations from these locations.</p>
<pre class="r"><code>#dapc analysis
dapc1 &lt;- dapc(pap_76_genlight, pop = clust$grp, n.pca = 40, n.da = 1)

#reshaping the data for plotting
dapc.results &lt;- as.data.frame(dapc1$posterior)
dapc.results$pop &lt;- pop(pap_76_genlight)
dapc.results$indNames &lt;- rownames(dapc.results)
dapc.results &lt;- melt(dapc.results)
colnames(dapc.results) &lt;-c(&quot;Original_Pop&quot;,&quot;Sample&quot;,&quot;Assigned_Pop&quot;,&quot;Posterior_membership_probability&quot;)

#memborship probability plot
memb_prob_plot &lt;- 
  ggplot(dapc.results, aes(x=Sample, y=Posterior_membership_probability, fill=Assigned_Pop)) + 
  geom_bar(stat=&#39;identity&#39;) + 
  facet_grid(~Original_Pop, scales = &quot;free&quot;) + scale_fill_brewer(palette = &quot;Accent&quot;) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 5), plot.title = element_text(face = &quot;bold&quot;, hjust = .5)) + 
  labs(title = &quot;Memborship Probability for H. Papilio,\n32 Sequence Reference&quot;, fill = &quot;Assigned Population&quot;, y = &quot;Memborship Probability&quot;)
memb_prob_plot</code></pre>
<p><img src="papilio_32_dapc_files/figure-html/unnamed-chunk-3-1.png" width="1056" /></p>
<p>We can perform the same analyses with the vcf file with variant information from <em>H. papilio</em> as well as <em>H. elegans</em>.</p>
<p>First, we’ll load and organize the data into the needed structure.</p>
<pre class="r"><code>#load vcf file as a vcfR object
pap_el_76_vcfR &lt;- read.vcfR(&quot;./Am_tu_Hp17_32seqs.fasta_bbmap_output_1_7_all_filtered.vcf&quot;)</code></pre>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 23
##   header_line: 24
##   variant count: 973
##   column count: 89
## 
Meta line 23 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 973
##   Character matrix gt cols: 89
##   skip: 0
##   nrows: 973
##   row_num: 0
## 
Processed variant: 973
## All variants processed</code></pre>
<pre class="r"><code>#convert vcfR object to genlight object
pap_el_76_genlight &lt;- vcfR2genlight(pap_el_76_vcfR)

#get vector of individual names
all_ind &lt;- pap_el_76_genlight %&gt;% 
  indNames () %&gt;% 
  substr(1,6) %&gt;% 
  factor(ordered=TRUE)

#filter population information for only those individuals, using same popoulation information file as used before (already loaded)
all_pop_new &lt;- master_pop %&gt;% 
  filter(lkh_number %in% all_ind)

#reorder population information to match order of individuals in genlight object
all_pop_new$lkh_number &lt;- factor(all_pop_new$lkh_number, levels = all_ind, ordered=TRUE)
all_reordered &lt;-  all_pop_new[order(all_pop_new$lkh_number),]

#change ANP to Acadia National Park for consistency
all_reordered &lt;- all_reordered %&gt;%
  mutate(location_new = ifelse(location_of_sample == &quot;ANP&quot;, 
                               &quot;Acadia National Park&quot;, 
                               location_of_sample))

#pull location vector                                  
location &lt;- all_reordered %&gt;% 
  pull(location_new) %&gt;% 
  factor(order=TRUE)

#make sure formatting of species names is consistent
species &lt;- all_reordered %&gt;% 
  mutate(&quot;species&quot; = case_when(
  str_detect(organism.x, &quot;elegans&quot;) ~ &quot;Hyalosphenia elegans&quot;,
  str_detect(organism.x, &quot;papilio&quot;) ~ &quot;Hyalosphenia papilio&quot;,
  TRUE ~ organism.x)) %&gt;% select(lkh_number, species)

#set population to location
pop(pap_el_76_genlight) &lt;- location

#add species information
rownames(species) &lt;- species$lkh_number
species &lt;- species %&gt;% select(species)
strata(pap_el_76_genlight) &lt;- species

#set ploidy to 1 (haploid)
ploidy(pap_el_76_genlight) &lt;- 1

#simplify names
indNames(pap_el_76_genlight) &lt;- substr(pap_el_76_genlight$ind.names,1,6)</code></pre>
<p>DAPC analysis for <em>H. papilio</em> and <em>H. elegans</em>.</p>
<p>First, we can force DAPC to use 2 clusters and see how the clustering compares to the species identification.</p>
<pre class="r"><code>#here, setting n.clust to 2 to see how dapc assignment differs from species assignment
clust &lt;- find.clusters(pap_el_76_genlight, max.n.clust = 15, n.pca = 40, n.clust = 2)

#compare to the actual species
table(strata(pap_el_76_genlight)$species, clust$grp )</code></pre>
<pre><code>##                       
##                         1  2
##   Hyalosphenia papilio 47  0
##   Hyalosphenia elegans  2 31</code></pre>
<pre class="r"><code>#dapc analysis
dapc1 &lt;- dapc(pap_el_76_genlight, pop = clust$grp, n.pca = 40, n.da = 1)

#reshaping the data for plotting
dapc.results &lt;- as.data.frame(dapc1$posterior)
dapc.results$species &lt;- strata(pap_el_76_genlight)$species
dapc.results$indNames &lt;- rownames(dapc.results)
dapc.results &lt;- melt(dapc.results)
colnames(dapc.results) &lt;-c(&quot;Original_Pop&quot;,&quot;Sample&quot;,&quot;Assigned_Pop&quot;,&quot;Posterior_membership_probability&quot;)

#memborship probability plot
memb_prob_plot &lt;- 
  ggplot(dapc.results, aes(x=Sample, y=Posterior_membership_probability, fill=Assigned_Pop)) + 
  geom_bar(stat=&#39;identity&#39;) + 
  facet_grid(~Original_Pop, scales = &quot;free&quot;) + scale_fill_brewer(palette = &quot;Accent&quot;) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 5), plot.title = element_text(face = &quot;bold&quot;, hjust = .5)) + 
  labs(title = &quot;Memborship Probability for H. Papilio,\n32 Sequence Reference&quot;, fill = &quot;Assigned Group&quot;, y = &quot;Memborship Probability&quot;)
memb_prob_plot</code></pre>
<p><img src="papilio_32_dapc_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<p>Then, we can allow DAPC to select the optimal number of clusters.</p>
<p>Looking at this result, we can see two main clusters within <em>H. papilio</em>, yet a higher degree of variability in cluster assignment within <em>H. elegans</em>. Again, it will be important to explore the effect of the reference on this result.</p>
<pre class="r"><code>clust &lt;- find.clusters(pap_el_76_genlight, max.n.clust = 15, n.pca = 40, n.clust = 9)

#compare to the actual populations
table(strata(pap_el_76_genlight)$species, clust$grp )</code></pre>
<pre><code>##                       
##                         1  2  3  4  5  6  7  8  9
##   Hyalosphenia papilio 19  0  0  0 13  0  0  0 15
##   Hyalosphenia elegans  0  2  5  3  0 19  2  2  0</code></pre>
<pre class="r"><code>#dapc analysis
dapc1 &lt;- dapc(pap_el_76_genlight, pop = clust$grp, n.pca = 40, n.da = 2)

#reshaping the data for plotting
dapc.results &lt;- as.data.frame(dapc1$posterior)
dapc.results$species &lt;- strata(pap_el_76_genlight)$species
dapc.results$pop &lt;- pop(pap_el_76_genlight)

dapc.results$indNames &lt;- rownames(dapc.results)
dapc.results &lt;- melt(dapc.results)
colnames(dapc.results) &lt;-c(&quot;Species&quot;, &quot;Location&quot;, &quot;Sample&quot;,&quot;Assigned_Pop&quot;,&quot;Posterior_membership_probability&quot;)

#memborship probability plot
memb_prob_plot &lt;- 
  ggplot(dapc.results, aes(x=Sample, y=Posterior_membership_probability, fill=Assigned_Pop)) + 
  geom_bar(stat=&#39;identity&#39;) + 
  facet_grid(Species ~Location, scales = &quot;free&quot;) + scale_fill_brewer(palette = &quot;Set3&quot;) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 5), plot.title = element_text(face = &quot;bold&quot;, hjust = .5)) + 
  labs(title = &quot;Memborship Probability for H. Papilio,\n32 Sequence Reference&quot;, fill = &quot;Assigned Group&quot;, y = &quot;Memborship Probability&quot;)

memb_prob_plot</code></pre>
<p><img src="papilio_32_dapc_files/figure-html/unnamed-chunk-6-1.png" width="1152" /></p>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
